quil-rs/src/program/analysis/stats/mod.rs

```diff
@@ -66,15 +68,41 @@ impl InstructionsSource for Program {
     }
 }
 
+impl<'p> InstructionsSource for BasicBlock<'p> {
```

Comment:
This is to fit the existing pattern, although would not be how I'd do it from scratch; see description

Reply:
It seems like `BasicBlock` is effectively what I was trying to capture with `InstructionsSource` anyway.